src/rust/cryptography-x509-validation/src/policy/mod.rs

```diff
@@ -234,15 +248,370 @@ impl<'a, B: CryptoOps> Policy<'a, B> {
+                    Some(ca::basic_constraints),
+                ),
+                // 5280 4.2.1.10: Name Constraints
+                // NOTE: MUST be critical in 5280, but CABF relaxes to MAY.
+                ExtensionPolicy::maybe_present(
+                    NAME_CONSTRAINTS_OID,
+                    Criticality::Agnostic,
+                    Some(ca::name_constraints),
+                ),
+                // 5280 4.2.1.11: Policy Constraints
+                ExtensionPolicy::maybe_present(POLICY_CONSTRAINTS_OID, Criticality::Critical, None),
+                // 5280: 4.2.1.12: Extended Key Usage
+                // NOTE: CABF requires EKUs in many non-root CA certs, but validators widely
+                // ignore this requirement and treat a missing EKU as "any EKU".
+                // We choose to be permissive here.
+                ExtensionPolicy::maybe_present(
+                    EXTENDED_KEY_USAGE_OID,
+                    Criticality::NonCritical,
+                    Some(ca::extended_key_usage),
+                ),
+            ]),
+            ee_extension_policies: Vec::from([
+                // 5280 4.2.1.1.: Authority Key Identifier
+                ExtensionPolicy::present(
+                    AUTHORITY_KEY_IDENTIFIER_OID,
+                    Criticality::NonCritical,
+                    None,
+                ),
+                // 5280 4.2.1.3: Key Usage
+                ExtensionPolicy::maybe_present(KEY_USAGE_OID, Criticality::Agnostic, None),
+                // CA/B 7.1.2.7.12 Subscriber Certificate Subject Alternative Name
+                ExtensionPolicy::present(
+                    SUBJECT_ALTERNATIVE_NAME_OID,
+                    Criticality::Agnostic,
+                    Some(ee::subject_alternative_name),
+                ),
+                // 5280 4.2.1.9: Basic Constraints
+                ExtensionPolicy::maybe_present(
+                    BASIC_CONSTRAINTS_OID,
+                    Criticality::Agnostic,
+                    Some(ee::basic_constraints),
+                ),
+                // 5280 4.2.1.10: Name Constraints
+                ExtensionPolicy::not_present(NAME_CONSTRAINTS_OID),
+                // CA/B: 7.1.2.7.10: Subscriber Certificate Extended Key Usage
+                // NOTE: CABF requires EKUs in EE certs, while RFC 5280 does not.
+                ExtensionPolicy::maybe_present(
+                    EXTENDED_KEY_USAGE_OID,
+                    Criticality::NonCritical,
+                    Some(ee::extended_key_usage),
+                ),
+            ]),
+        }
+    }
+
+    fn permits_basic(&self, cert: &Certificate<'_>) -> Result<(), ValidationError> {
+        let extensions = cert.extensions()?;
+
+        // CA/B 7.1.1:
+        // Certificates MUST be of type X.509 v3.
+        if cert.tbs_cert.version != 2 {
+            return Err(ValidationError::Other(
+                "certificate must be an X509v3 certificate".to_string(),
+            ));
```

Comment:
@reaperhulk @alex thoughts on removing this check entirely? It's now hard to get coverage for since we've moved the SAN check into `permits_leaf`, which runs ahead of this and will transitively eliminate non-v3 certs by looking for an extension.

Reply:
Reordered to get this coverage.